mlflow/transformers/__init__.py

```diff
@@ -2581,6 +2588,63 @@ def _convert_cast_lists_from_np_back_to_list(data):
                     parsed_data.append(entry)
             return parsed_data
 
+    @staticmethod
+    def is_base64_image(image):
+        """Check whether input image is a base64 encoded"""
+
+        try:
+            return base64.b64encode(base64.b64decode(image)).decode("utf-8") == image
```

Comment:
without the decode part, they are not matching

Reply:
I think if you don't decode the image when you read it, then it matches:
image = base64.b64encode(read_image("cat_image.jpg"))
base64.b64encode(base64.b64decode(image)) == image

```
>>> with open("/Users/serena.ruan/Documents/repos/mlflow/tests/datasets/cat.png", "rb") as f:
...     image = f.read()
... 
>>> import base64
>>> image = base64.b64encode(image)
>>> base64.b64encode(base64.b64decode(image)) == image
```

Reply:
if isinstance(image, str):
        if image.startswith("http://") or image.startswith("https://"):
            # We need to actually check for a real protocol, otherwise it's impossible to use a local file
            # like http_huggingface_co.png
            image = PIL.Image.open(requests.get(image, stream=True, timeout=timeout).raw)
        elif os.path.isfile(image):
            image = PIL.Image.open(image)
        else:
            if image.startswith("data:image/"):
                image = image.split(",")[1]

            # Try to load as base64
            try:
                b64 = base64.b64decode(image, validate=True)
                image = PIL.Image.open(BytesIO(b64))
            except Exception as e:
                raise ValueError(
                    f"Incorrect image source. Must be a valid URL starting with `http://` or `https://`, a valid path to an image file, or a base64 encoded string. Got {image}. Failed with {e}"
                )


transformers accepts the base64 image in a string format. It's not accepting direct b64 encoded image format which is in bytes.